src/app/shared/auth/auth.service.ts

```diff
@@ -82,7 +82,12 @@ export class AuthService implements OnDestroy {
         return timer(Math.min(refreshIn, maxDelay), 1*MINUTES).pipe(map(() => auth));
       }),
       switchMap(auth =>
-        this.authHttp.refreshAuth(auth)
+        this.authHttp.refreshAuth(auth).pipe(
+          catchError(() => {
+            this.invalidToken(auth);
```

Comment:
so, side effect and returning empty? 🤨
Why not just handling the errors in the subscribe?

And so you are considering all error as "the token is invalid, let's drop it disconnect the user, ..."... disregarding what the error is? That's not good. Here we try to renew a token before it expires (in the normal case)... so it is not because we receive an error that the token in invalid... if it is still valid and we are just unable to extend (refresh) it, let's just use it to the end instead of dropping it.

Reply:
If I handle the error in the subscribe, the subscription will be closed and your token won't get refreshed anymore next times.
To give an example with the case you encountered:
1. You come back after a computer suspend, this subscription falls into the error case.
2. The error fn is executed, and it does an `invalidToken` (if appropriate)
3. After the given delay, no refresh will occur because the subscription is now closed because of the previous error. To me, the expected behavior is that the platform should keep on refreshing tokens after the given delay.

About the side effect part: fixed.
About not filtering errors: fixed.